### DISCUSSION TITLE
Ask: Update composer config for new plugin installed by user from wp-admin panel

### DISCUSSION THREAD

**Discussion by brm-ryd** [composer/composer#10774] [Q&A / Support] [answered]
Hi,

Sorry, I'm wondering if its possible to automatically update the composer config file when there's a new plugin by user who install plugins from wp-admin panel for example ?

- Do you think its possible if the wordpress directory part of git and perhaps i can tweak something to scan / detect then update the composer file ? im currently try to read / test something with like hooks though 
- Or maybe there are some ways to achieve this ? 

Many thanks in advance,

**Comment by Seldaek** [comment]
I'm not sure what you are asking, but yes I would think it's doable, but I am not sure about the wordpress part as I have not much knowledge on that front. How are plugins installed exactly? How do you run Composer?

**Comment by brm-ryd** [comment]
For example if some admin user access the wordpress admin panel (GUI) [https://themeisle.com/blog/how-to-install-a-wordpress-plugin/](https://themeisle.com/blog/how-to-install-a-wordpress-plugin/)
Cmiiw, by default composer will just install / do config based on config file isnt ? at first deploy i ran the composer either by command in server / through container, but if somehow when there's an update by other user admin which not through composer instead the user install new plugin from GUI i'd like to at least make composer can update the content of config for that new plugin without adding the config manually i mean 

Wordpress usually stored plugins in wp-content/plugins/ - do you think i can achieve that purpose ?

Sorry & many thanks for your response

**Reply by Seldaek** [reply]
I have no idea if you can still use the admin to install plugins, I doubt this will play well with a composer-managed WP install. I guess you're using something like Bedrock? Maybe best to ask for support there..

**Comment by brm-ryd** [comment, marked as answer]
Sorry, no at this moment i'm not yet use Bedrock (i plan to try bedrock though), just thought if i can make something to update the content of list plugin in the composer when user install new plugin manually. I'll also check with Bedrock for this later. Many thanks @Seldaek (hope later i can open up discussion again in this Composer repo)
